src/workerd/api/sql.c++

```diff
@@ -22,6 +22,27 @@ jsg::Ref<SqlStorage::Statement> SqlStorage::prepare(jsg::Lock& js, kj::String qu
   return jsg::alloc<Statement>(sqlite->prepare(*this, query));
 }
 
+double SqlStorage::getDatabsaeSize() {
+  int64_t pages = execMemoized(pragmaPageCount, "PRAGMA page_count;").getInt64(0);
+  return pages * getPageSize();
+}
+
+double SqlStorage::getVoluntarySizeLimit() {
+  int64_t pages = execMemoized(pragmaGetMaxPageCount, "PRAGMA max_page_count;").getInt64(0);
+  return pages * getPageSize();
+}
+
+void SqlStorage::setVoluntarySizeLimit(int64_t value) {
+  JSG_REQUIRE(value > 0, TypeError, "Size limit must be a positive integer.");
+
+  uint64_t pages = value / getPageSize();
```

Comment:
I suppose you could argue both ways but perhaps we should add 1 and round up?

Reply:
I don't think so. I interpret the limit as meaning "The database shall not be allowed to exceed this size." Rounding up means that the database is allowed to exceed the limit (by a partial page). Rounding down does not prohibit any valid databases that comply with the limit, since databases cannot have partial pages.